plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineBundle.kt

```diff
@@ -85,6 +89,61 @@ open class GenerateGramineBundle @Inject constructor(
         }
     }
 
+    private fun findJavaModules(): String {
+        val enclaveJar = outputDir.file(GRAMINE_ENCLAVE_JAR).get().asFile.absolutePath
+
+        val dependentModules = execCommand(
+            "jdeps",
+            "--print-module-deps",
+            "--ignore-missing-deps",
+            enclaveJar
+        ).run { trimEnd().replace(System.lineSeparator(), "").split(",") }
+        // TODO: Provide mechanism to allow the user to add modules needed when using reflection
+        val defaultModules = listOf("java.base", "java.desktop", "java.sql", "jdk.crypto.ec")
+        return (defaultModules + dependentModules).distinct().joinToString(separator = ",")
+    }
+
+    private fun createCustomJDK() {
+        val jlinkOutputPath = "${outputDir.get()}/$JLINK_OUTPUT_DIRECTORY"
```

Comment:
something like `outputDir.get().asFile.toPath().resolve(JLINK_OUTPUT_DIRECTORY)` to keep the type as a `Path`.

Reply:
Do we have a nice recursive delete in Kotlin? I see that it is experimental, but that is the main reason of why I wanted to use File here

Reply:
`.toFile().deleteRecursively()` is your best bet i think